lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -1822,22 +1830,24 @@ void ByteCodeGenerator::InitScopeSlotArray(FuncInfo * funcInfo)
 
         funcInfo->GetBodyScope()->ForEachSymbol(setPropIdsForScopeSlotArray);
 
-        if (funcInfo->thisScopeSlot != Js::Constants::NoRegister)
+        bool isSplitScope = funcInfo->paramScope && !funcInfo->paramScope->GetCanMergeWithBodyScope();
+
```

Comment:
Can the if statements that follow this line be grouped so that each one doesn't have to check !isSplitScope?